wikibot/bot.py

```diff
@@ -0,0 +1,87 @@
+import requests
+
+import misaka
+from wikibot.github_wrapper import GithubWikiWrapper
+
+
+class Wikibot(GithubWikiWrapper):
+    wiki_root = 'http://wiki.citi.org.br'
+    api_root = wiki_root + '/api.php'
+    session = requests.Session()
+
+    def __init__(self, username=None, password=None, page=None):
+        self.username = username
+        self.password = password
+        self.page = page
+        self._log_in()
+
+    @property
+    def login_token(self):
+        response = self.session.post(self.api_root, data={
+            'format': 'json',
+            'action': 'login',
+            'lgname': self.username
+        })
+        response.raise_for_status()
+        return response.json()['login']['token']
+
+    @property
+    def edit_token(self):
+        response = self.session.get(self.api_root, params={
+            'action': 'query',
+            'meta': 'tokens',
+            'format': 'json',
+            'prop': 'info'
+        })
+        return response.json()['query']['tokens']['csrftoken']
+
+    def _log_in(self):
+        response = self.session.post(self.api_root, data={
+            'format': 'json',
+            'action': 'login',
+            'lgname': self.username,
+            'lgpassword': self.password,
+            'lgtoken': self.login_token
+        })
+        if not response.json()['login']['result'] == 'Success':
+            raise RuntimeError(response.json()['login']['reason'])
+        print('Successfully logged in as {user}!'.format(user=self.username))
+        return response
+
+    def edit_page(self, content=None):
+        """ Overwrite existing page with the content you provide (supports Wikicode and basic HTML) """
+        if not content != None:
```

Comment:
Can't this line (and some similar one) be written as `if something is None`? It'd be a bit more legible imo.

Reply:
There's a difference between `==` and `is` in Python, [check here](https://stackoverflow.com/a/5782243/4526204). Seems like `is` is more strict, but I'm gonna take a look into it.

Reply:
You're absolutely correct in the differences of what `==` and `is` compare. `None` is a kind of special case because it seems to work with both (although it seems `is` comparison is a [bit faster](https://stackoverflow.com/questions/3257919/what-is-the-difference-between-is-none-and-none?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa#comment3371949_3257957) than `==`). Also, [PEP 8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations) recommends that all `None` comparisons are always made using `is`.

Reply:
Nice! Got it. Thank you so much about the references. I'm gonna edit it.